src/Core/Mesh/TriangleMesh.hpp

```diff
@@ -19,67 +17,89 @@ namespace Core
 /// The VertexAttribManager allows to dynammicaly add VertexAttrib per Vertex
 /// See MeshUtils for geometric functions operating on a mesh.
 /// Points and Normals are always present, accessible with
-/// points() and normals()
+/// points() and normals().
 /// Other attribs could be added with attribManager().addAttrib() and
 /// accesssed with attribManager().getAttrib()
-class TriangleMesh
-{
+/// Note: attribute names "in_position" and "in_normal" are reserved.
+class TriangleMesh {
   public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
-    using PointAttribHandle  = AttribHandle<Vector3>;
+    using PointAttribHandle = AttribHandle<Vector3>;
     using NormalAttribHandle = AttribHandle<Vector3>;
-    using FloatAttribHandle  = AttribHandle<float>;
-    using Vec2AttribHandle   = AttribHandle<Vector2>;
-    using Vec3AttribHandle   = AttribHandle<Vector3>;
-    using Vec4AttribHandle   = AttribHandle<Vector4>;
-    using Face               = VectorNui;
+    using FloatAttribHandle = AttribHandle<float>;
+    using Vec2AttribHandle = AttribHandle<Vector2>;
+    using Vec3AttribHandle = AttribHandle<Vector3>;
+    using Vec4AttribHandle = AttribHandle<Vector4>;
+    using Face = VectorNui;
 
     /// Create an empty mesh.
     inline TriangleMesh() { initDefaultAttribs(); }
-    /// Copy constructor and assignment operator
-    /// \warning No deep copy of attributes !
+
+    /// Copy constructor and assignment operator are pure shallow copy of attributes.
+    /// For deep copy of attributes, use the partialCopy or fullCopy methods.
+    /// Note: Directly using the AttribManager is highly discouraged.
     TriangleMesh( const TriangleMesh& ) = default;
     TriangleMesh& operator=( const TriangleMesh& ) = default;
 
+    /// Copy only the required attributes (deep copy).
+    /// The position and normal attributes are always copied, no need to provide their handles.
+    /// Note: The original handles are not valid for the mesh copy.
+    template <typename... Handles>
+    void partialCopy( const TriangleMesh& input, Handles... attribs );
```

Comment:
The name of the method is to me very confusing.
I would prefer something like `copyAttributes`, which express a bit more what's happening when the method is called.
Also, I don't see why the positions and normals are necessary copied. I could understand this for the positions, but in some cases you might have meshes without normals.